src/wallet/wallet.h

```diff
@@ -455,6 +455,9 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     void SetSpentKeyState(WalletBatch& batch, const uint256& hash, unsigned int n, bool used, std::set<CTxDestination>& tx_destinations) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
 
     std::vector<OutputGroup> GroupOutputs(const std::vector<COutput>& outputs, const CoinSelectionParams& coin_sel_params, const CoinEligibilityFilter& filter, bool positive_only) const;
+    //! Whether this address has been used.
+    void SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, uint256 hash) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
Here and in the definition in the .cpp file, `uint256 hash` ought to be passed by reference to const, as it is not a cheaply passed type (see usage elsewhere in the codebase). 
```suggestion
    void SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, const uint256& hash) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const

Reply:
Thanks @jonatack, this should now be resolved in bb4790816d84f850367c99134ea6a16659fbb180